bukkit/src/main/java/io/ibj/mcauthenticator/engine/events/ChatEvent.java

```diff
@@ -28,7 +28,7 @@ public void onPlayerChat(AsyncPlayerChatEvent event) {
             Iterator<Player> recipients = event.getRecipients().iterator();
             while(recipients.hasNext()) {
                 User u = instance.getCache().get(recipients.next().getUniqueId());
-                if(!u.authenticated()) recipients.remove();
+                if(u != null && !u.authenticated()) recipients.remove();
```

Comment:
Yes, if the message doesn't arrive anyway, why are we sending it to them? Is there some other consideration we need? I feel like it actually falls into bukkit behavior, where we are moving an 'erroneous' bukkit player.

I also think this way because it will be easier to detect a horrible problem with the User cache system if the person who's User loaded wrong didn't get chat messages.